src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediatePreparedStatement.java

```diff
@@ -118,36 +87,22 @@ public void execute() {
    * @param parameterFormatCodes A list of the format of each parameter.
    * @param resultFormatCodes A list of the desired format of each result.
    * @return An Intermediate Portal Statement (or rather a bound version of this statement)
-   * @throws SQLException If the binding fails.
    */
   public IntermediatePortalStatement bind(
-      byte[][] parameters, List<Short> parameterFormatCodes, List<Short> resultFormatCodes)
-      throws SQLException {
-    IntermediatePortalStatement portal =
-        new IntermediatePortalStatement(
-            this.connection.prepareStatement(this.sql),
-            this.sql,
-            this.parameterCount,
-            this.parameterIndexToPositions,
-            this.connection);
+      byte[][] parameters, List<Short> parameterFormatCodes, List<Short> resultFormatCodes) {
+    IntermediatePortalStatement portal = new IntermediatePortalStatement(this.sql, this.connection);
     portal.setParameterFormatCodes(parameterFormatCodes);
     portal.setResultFormatCodes(resultFormatCodes);
-    if (parameters.length != parameterCount) {
-      throw new IllegalArgumentException(
-          "Wrong number of parameters for prepared statement, expected: "
-              + parameterCount
-              + " but got: "
-              + parameters.length);
-    }
+    Statement.Builder builder = Statement.newBuilder(sql);
     for (int index = 0; index < parameters.length; index++) {
       short formatCode = portal.getParameterFormatCode(index);
       int type = this.parseType(parameters, index);
-      for (Integer position : parameterIndexToPositions.get(index)) {
```

Comment:
We don't need to translate back-and-forth between positional parameters (?) and indexed parameters ($1, $2, ...) anymore, which removes the requirement for the translation map `parameterIndexToPositions`.